homekit/model/services/service_types.py

```diff
@@ -84,9 +84,27 @@ def get_short(self, item):
         return 'Unknown Service: {i}'.format(i=orig_item)
 
     def get_uuid(self, item_name):
-        if item_name not in self._services_rev:
-            raise Exception('Unknown service name')
-        short = self._services_rev[item_name]
+        """
+        Returns the full length UUID for either a shorted UUID or textual characteristic type name. For information on
+        full and short UUID consult chapter 5.6.1 page 72 of the specification. It also supports to pass through full
+        HomeKit UUIDs.
+
+        :param item_name: either the type name (e.g. "public.hap.characteristic.position.current") or the short UUID or
+                          a HomeKit specific full UUID.
+        :return: the full UUID (e.g. "0000006D-0000-1000-8000-0026BB765291")
+        :raises KeyError: if the input is neither a short UUID nor a type name. Specific error is given in the message.
+        """
+        orig_item = item_name
```

Comment:
should this function also be able to handle lower case UUIDs? Just in case these are fed into it?

Reply:
If the fix is applied inside controller.py - then there is no chance of passing here lowercase UUIDS.
They all will be normalized to full uppercase uuids long before this function call.

Reply:
Ok, maybe there will be need to use this function externally without access to data from device.
Then it makes sense to also normalize here?

Reply:
Since UUIDs seem to be the same for upper and lower case version, I would tend to have the code be able to handle both versions as input as well.